qiskit/transpiler/passes/scheduling/base_scheduler.py

```diff
@@ -0,0 +1,269 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base circuit scheduling pass."""
+
+from typing import Dict
+from qiskit.transpiler import InstructionDurations
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.passes.scheduling.time_unit_conversion import TimeUnitConversion
+from qiskit.dagcircuit import DAGOpNode, DAGCircuit
+from qiskit.circuit import Delay, Gate
+from qiskit.circuit.parameterexpression import ParameterExpression
+from qiskit.transpiler.exceptions import TranspilerError
+
+
+class BaseScheduler(TransformationPass):
```

Comment:
I'm a bit worried about the renaming this because what if someone wrote a downstream scheduler pass that was different from alap and asap for some custom application. If we rename this class we'd break those people right? Maybe we should rename the qiskit/transpiler/passes/scheduling/schedling/base_scheduler.py version to `BaseSchedulerAnalysis` or something instead?

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/7835/commits/f992ea4725cbaaeba54f7f967b728c4c197ee21f